### ISSUE TITLE
Exploding stuff post-gameover

### ISSUE THREAD

**Issue by sprunk** [ZeroK-RTS/Zero-K#1648]
Decide whether loser's stuff should explode on resign as it currently does.

Pros:
 * makes it obvious who won
 * does not give much incentive to stay (recycles people into the next game faster)
 * legacy (ctrl+AD was the honourable way to resign in ye olde days)

Cons:
 * the winner doesn't get to discover interesting stuff that went unscouted (hidden expansions, attempts at superweapons)
 * newbs that don't pay attention might find it weird that their stuff just randomly blew up (though admittedly simply losing control isn't much better)
 * if the dedicated host does not immediately exit (so that people can chat after the game) the winners get to blow stuff up themselves if they want to

**Comment by Licho1** [comment]
Yeah I think we should remove mass destruction at the end.

**Comment by Anarchid** [comment]
It could be nice to still indicate clearly who lost, perhaps by giving all loser units to Gaia, turning them gray; or otherwise changing their appearance to some low-status color.

I'm contra letting winners blow more stuff up because this makes it unclear that game has actually ended. IMO, no units should be able to die once resign has occured. The frozen game board should be available for inspection in a linger state.

**Comment by ArchShaman** [comment]
Lose control as a loss condition? It's already a modoption IIRC.

**Comment by sprunk** [comment]
Probably. In any case it's a one-liner.

**Comment by mtroyka** [comment]
meh, the only thing I really dislike about the end-of-game explosions is that it's pretty much instantaneous which can cause an infinite lag spike as lua catches up with 9001 unit destroyed calls. It would probably work a lot better if the self-d commands were staggered, by ie killing only one unit per frame.

I think if you want to show the winning team the end-of-game state there are potentially better ways of doing it (ghost site, for example) than to leave zombies.

**Comment by Anarchid** [comment]
> the only thing I really dislike about the end-of-game explosions is that it's pretty much instantaneous which can cause an infinite lag spike

In recent versions it staggers the destruction over several frames, which i find significantly decreases the slowdown at least for me - to the point where i don't notice it.

**Comment by let4be** [comment]
Can we please remove this annoying explode all the enemy feature?
I want to check what enemy had before quitting, do I have to watch game till the end on replay to find out some aspects?... this may take hours in long game

**Comment by CrazyEddieTK** [comment]
Mass destruction is a very aesthetically-pleasing way to mark the end of the game. Just a victory screen alone seems anti-climactic.

Perhaps this could be made an option, with "blow everything up at the end" being the default, and allowing seasoned pros like @let4be to disable it if they choose? Since this only happens after game-end, it could be a widget instead of a gadget; the game no longer needs to stay in sync between players.

Alternately, come up with something else suitably dramatic as the end-of-game marker.

**Comment by Anarchid** [comment]
Explosions are synced. You cannot disable explosions for yourself only i think.

**Comment by sprunk** [comment]
Technically you can but it desyncs the game so is a bad idea (regardless of being only after game over).

**Comment by CrazyEddieTK** [comment]
This might be ambitious, but how about blowing up all the units like usual, but drawing ghosts of them first and leaving the ghosts displayed indefinitely?

Best of both worlds. Dramatic ending, but everyone can still see what everyone had. It's even better, because the game won't continue to play on autopilot; it'll truly reflect what was still on the field when the losing side resigned. Perhaps also autopause the game so that _both_ sides stop autopiloting, and disable unpausing. 

So:
 - Blow up the losing side
 - Wait a few seconds so the rubble stops bouncing
 - Permanently pause the game
 - Draw ghosts for all the destroyed losing-side units

Everyone's happy, and the game has some unique and fun features.

**Comment by Licho1** [comment]
I don't think that 2s lag is dramatic... you dont usually see much of the stuff exploding and it stutters badly.

I think it would be enough if for example all commanders exploded .. or just add some sound/music (different for victory/defeat)

**Comment by sprunk** [comment]
There already is a different music for victory and defeat (though the victory music is sad and not epic, could be better)

**Comment by Anarchid** [comment]
Doing things to loser's team colors could work.

**Comment by GoogleFrog** [comment]
I like things exploding when the game ends. In singleplayer units would keep fighting indefinitely while you look at the game over screen. If your opponent resigns and their units are saved you don't get the chance to destroy their base. At least when things explode you get to destroy their base.

You can usually see what your opponents had by looking at the explosion effects/debri.

**Comment by Licho1** [comment]
Looking at explosions/debri to understand former battlefield is likely a skill aquired only after hundreds of hours..

**Comment by sprunk** [comment]
21855bfdb4022d78487a79881efb859dac05cc14

**Comment by let4be** [comment]
probably pause the game before exploding?

**Comment by GoogleFrog** [comment]
Show ghosts depicting what your opponents had via a toggleable overlay?

**Comment by Licho1** [comment]
Nah ghosts are not sufficient.. often you are interested in full state.. units, projectiles, explosions...

Imo ghosts are not worth considering.

**Comment by CrazyEddieTK** [comment]
>  often you are interested in full state.. units, 

Units could be ghosted as well, could they not?

> projectiles, explosions...

You want to see what projectiles are in flight and what explosions are in progress at the moment a resign vote passes?

**Comment by sprunk** [comment]
>Units could be ghosted as well, could they not?

They would need to be more advanced than regular gameplay ghosts, showing health, stockpile and such.

**Comment by CrazyEddieTK** [comment]
Relatedly: when the game ends, perhaps we should leave the  simulation running so that (for example) planes keep circling instead of being frozen in midair, but otherwise "suspend" the game: give all units a Stop order, put all units on Hold Fire and Hold Position, turn off things like Starlights and Outlaws so they don't keep doing damage, manually reset metal and energy stockpiles every frame to the value they had at the end of the game, stop updating the gametime clock display, etc. And of course make the units uncontrollable by making everyone spectators (like normal).

That way (for example) factories don't keep churning out units that just gather around the waypoint like they're waiting for a concert to start or something.

**Comment by sprunk** [comment]
That kinda already happens, after a few seconds the dedicated server stops and clients freeze.

**Comment by sprunk** [comment]
Duplicate of #4043
